guides/advanced_user_guide.md

```diff
@@ -0,0 +1,97 @@
+
+### Defining Your Own Tekton Custom Task on KFP-Tekton
+1. To define your own Tekton custom task on KFP-Tekton, first define a unique image name for classifying your containerOp definition as a custom task and update the list of registered custom task images.
+    ```python
+    MY_CUSTOM_TASK_IMAGE_NAME = "veryunique/image:latest"
+    from kfp_tekton.tekton import TEKTON_CUSTOM_TASK_IMAGES
+    TEKTON_CUSTOM_TASK_IMAGES = TEKTON_CUSTOM_TASK_IMAGES.append(MY_CUSTOM_TASK_IMAGE_NAME)
+    ```
+2. Define your own Tekton custom task as a containerOp following the below constraints:
+    ```python
+    ConditionOp = dsl.ContainerOp(
+            name="any-name",
+            image=MY_CUSTOM_TASK_IMAGE_NAME,
+            command=["any", "command"],
+            arguments=["--apiVersion", "custom_task_api_version",
+                       "--kind", "custom_task_kind",
+                       "--name", "custom_task_name",
+                       "--other_custom_task_argument_keys", custom_task_argument_values],
+            file_outputs={"other_custom_task_argument_keys": '/anypath'}
+        )
+    # Annotation to tell the Argo controller that this containerOp is for specific Tekton runtime only.
+    ConditionOp.add_pod_annotation("valid_container", "false")
+    ```
+    - **name**: Name for the custom task, it can be anything with a valid Kubernetes name.
+    - **image**: Identifier for your custom task. This is the value you defined above in step 1. If you want to use a custom task with CEL condition support (such as using literal results from CEL custom task for Tekton whenExpression), then use `CEL_EVAL_IMAGE` instead.
+    - **arguments**: Define the key value pairs for the custom task definition. Below are the mandatory fields
+      - **--apiVersion**: Kubernetes API Version for your custom task CRD.
+      - **--kind**: Kubernetes Kind for your custom task CRD.
+      - **--name**: Kubernetes Resource reference name for your custom task CRD.
+
+      Then, you can add any extra arguments to map as the custom task CRD's `spec.params` fields.
+    - **--file_outputs**: List of Tekton results from the custom task. Since the custom task spec doesn't have Tekton results in static compile time, users need to explicitly define the custom task outputs to avoid mismatching between the custom task I/O.
+
+
+### Custom task with Conditions
+When using custom task with conditions, the compiled workload is slightly different from Argo because we are not wrapping condition blocks as sub-pipeline. The [flip-coin-custom-task](/samples/flip-coin-custom-task) example describes how custom tasks are used with KFP conditions.
```

Comment:
```suggestion
When using a custom task with conditions, the compiled workload is slightly different from Argo because we are not wrapping condition blocks as a sub-pipeline. The [flip-coin-custom-task](/samples/flip-coin-custom-task) example describes how custom tasks are used with KFP conditions.
```